docs/advanced/prompt_suggestions.md

```diff
@@ -0,0 +1,237 @@
+
+!!! success "YANG-driven path suggestions"
+    `gnmic` is now capable of reading and processing YANG modules to enable live path auto-suggestions
+
+### YANG processing
+For the YANG-completion feature to work its absolutely imperative for `gnmic` to successfully parse and compile the YANG models.
+
+The [`prompt`](../cmd/prompt.md) command leverages the [`--file`](../cmd/prompt.md#file) and [`--dir`](../cmd/prompt.md#dir) flags to select the YANG models for processing.
+
+
+With the `--file` flag a user specifies a file path to a YANG file or a directory of them that `gnmic` will read and process. If it points to a directory it will be visited recursively reading in all `*.yang` files it finds.
+
+The `--dir` flag also points to a YANG file or a directory and indicates which additional YANG files might be required. For example, if the YANG modules that a user specified with the `--file` flag import or include modules that were not part of the path specified with `--file`, they need to be added with the `--dir` flag.
+
+The [Examples](#examples) section provide some good practical examples on how these two flags can be used together to process the YANG models from different vendors.
+
+### Understanding path suggestions
+When `gnmic` provides a user with the path suggestions it does it in a very unique and intellectual way.
+
+![path suggestions](https://gitlab.com/rdodin/pics/-/wikis/uploads/d3815b474605765989d136753c0f9c87/image.png)
+
+First, it understands in what part of the tree a user currently is and suggests only the next possible elements.
+
+Additionally, the suggested next path elements will be augmented with the information extracted from the YANG model, such as:
+
+* element description, as given in the YANG `description` statement for the element
+* element configuration state (`rw` / `ro`), as defined in section [4.2.3 of RFC 7950](https://tools.ietf.org/html/rfc7950#section-4.2.3).
+* node type:
+    * The containers and lists will be denoted with the `[+]` marker, which means that a user can type `/` char after them to receive suggestions for the nested elements.
+    * the `[⋯]` character belongs to a leaf-list element.
+    * an empty space will indicate the leaf element.
+
+### Examples
+The examples in this section will show how to use the `--file` and `--dir` flags of the [`prompt`](../cmd/prompt.md) command with the YANG collections from different vendors and standard bodies.
+
+#### Nokia SR OS
+YANG repo: [nokia/7x50_YangModels](https://github.com/nokia/7x50_YangModels)
+
+Clone the repository with Nokia YANG models and checkout the release of interest:
+
+```
+git clone https://github.com/nokia/7x50_YangModels
+cd 7x50_YangModels
+git checkout sros_20.7.r2
+```
+
+Start `gnmic` in prompt mode and read in the nokia-combined YANG modules:
+
+```
+gnmic --file YANG/nokia-combined \
+      --dir YANG \
+      prompt
+```
+
+This will enable path auto-suggestions for the entire tree of the Nokia YANG models.
```

Comment:
Nokia SR OS

Reply:
added in a0d15077bf6f8e91f99d53aa7f65599e3a72e450